### PR TITLE
ref: Move events specific processor code into EventsProcessor

### LINKED ISSUE #336: Move events Kafka message processing into the EventsProcessor class
As of #329  we will have a processor class instantiated by the dataset. This should replace the processor.py file (a lot of independent functions.
We can encapsulate that logic into EventsProcessor and remove some cumbersome column list passed around.

### PR DESCRIPTION
**Pull request by lynnagara**
Closes #336

### CODE REVIEW

File: snuba/consumer.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ def delivery_callback(self, error, message):
     def flush_batch(self, batch):
         """First write out all new INSERTs as a single batch, then reproduce any
         event replacements such as deletions, merges and unmerges."""
+        processor = self.__dataset.get_processor()
```
Comment (fpacifici) [inline snuba/consumer.py]: unused ?

File: tests/datasets/test_events_processor.py [hunk 2]
```diff
@@ -4,55 +4,52 @@
...
 
 
-class TestProcessor(BaseTest):
-    def setup_method(self, test_method):
-        super(TestProcessor, self).setup_method(test_method, 'events')
-        self.event = get_event()
-
+class TestEventsProcessor(BaseEventsTest):
     def test_simple(self):
-        _, processed = process_message(self.dataset.get_promoted_tag_columns(), self.event)
```
Comment (fpacifici) [inline tests/datasets/test_events_processor.py]: do we still need to expose get_promoted_tag_columns on the dataset ? Or can we jsut remove the method or making it private ?

File: snuba/datasets/events_processor.py [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+        self._extract_extra_contexts(processed, contexts)
+        self._extract_extra_tags(processed, tags)
+
+        exception = data.get('exception', data.get('sentry.interfaces.Exception', None)) or {}
+        stacks = exception.get('values', None) or []
+        self._extract_stacktraces(processed, stacks)
+
+        return processed
+
+    def _extract_required(self, output, message):
```
Comment (fpacifici) [inline snuba/datasets/events_processor.py]: unless you want subclassses yo use this (which should not be the case) you can make them "private" (which is not really private but anyway ...) with two underscores.
Or if you need them in tests just leave them public ?

File: snuba/datasets/events_processor.py [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+        output['exception_frames.colno'] = frame_colnos
+        output['exception_frames.lineno'] = frame_linenos
+        output['exception_frames.stack_level'] = frame_stack_levels
+
+
+class EventTooOld(Exception):
+    pass
+
+
+def _as_dict_safe(value):
```
Comment (fpacifici) [inline snuba/datasets/events_processor.py]: The methods here and below are likely useful for others processors as well. I would suggest you leave them where they were (I was planning to use unicodify for the groupedmessage for  example, but it turned out I did not need strings.

File: snuba/datasets/events_processor.py (hunk 3 above)
Comment (lynnagara) [reply snuba/datasets/events_processor.py]: It's not really supposed to be public, but tests are testing this directly right now. Will just make it public then.

File: snuba/consumer.py (hunk 1 above)
Comment (lynnagara) [reply snuba/consumer.py]: It's needed for lines 69/71

File: tests/datasets/test_events_processor.py (hunk 2 above)
Comment (lynnagara) [reply tests/datasets/test_events_processor.py]: It's used in one last place right now, will remove it completely later in https://github.com/getsentry/snuba/pull/346

**Pull request merged**